Index: app/src/main/java/com/example/bkkinfoplus/ui/AlertListFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/bkkinfoplus/ui/AlertListFragment.java	(revision d7adb6ab2f51be26abaceb5b6137461cdfe9a57e)
+++ app/src/main/java/com/example/bkkinfoplus/ui/AlertListFragment.java	(revision )
@@ -16,7 +16,7 @@
 import com.android.volley.Response;
 import com.android.volley.VolleyError;
 import com.example.bkkinfoplus.Alert;
-import com.example.bkkinfoplus.AlertLoader;
+import com.example.bkkinfoplus.FutarApiClient;
 import com.example.bkkinfoplus.BkkInfoApplication;
 import com.example.bkkinfoplus.R;
 import com.example.bkkinfoplus.Route;
@@ -41,7 +41,8 @@
     private AlertAdapter mAlertAdapter;
     private SwipeRefreshLayout mSwipeRefreshLayout;
 
-    @Inject AlertLoader mAlertLoader;
+    @Inject
+    FutarApiClient mFutarApiClient;
 
     private long mLastUpdate;
     private static final int REFRESH_THRESHOLD_SEC = 30;
@@ -96,8 +97,8 @@
     private void initRefresh() {
         Context applicationContext = this.getActivity().getApplicationContext();
 
-        if (mAlertLoader != null) {
-            mAlertLoader.fetchAlertList(applicationContext, this, this);
+        if (mFutarApiClient != null) {
+            mFutarApiClient.fetchAlertList(applicationContext, this, this);
         }
 
         mLastUpdate = new GregorianCalendar().getTimeInMillis();
@@ -130,9 +131,9 @@
     public void onResponse(JSONObject response) {
         List<Alert> alerts = null;
         try {
-           alerts = mAlertLoader.parseAlertList(response);
+           alerts = mFutarApiClient.parseAlertList(response);
             //TODO: refactor
-            mAlertLoader.updateRoutes(response);
+            mFutarApiClient.updateRoutes(response);
         } catch (Exception ex) {
             Toast.makeText(getActivity(), R.string.error_list_display, Toast.LENGTH_LONG).show();
         }
@@ -189,7 +190,7 @@
 
             if (alert.getRouteIds() != null) {
                 for (String routeId : alert.getRouteIds()) {
-                    Route route = mAlertLoader.getRoute(routeId);
+                    Route route = mFutarApiClient.getRoute(routeId);
 
                     if (route == null) {
                         // Replacement routes are filtered out at the parse stage,
Index: app/src/main/java/com/example/bkkinfoplus/AlertLoader.java
===================================================================
--- app/src/main/java/com/example/bkkinfoplus/AlertLoader.java	(revision d7adb6ab2f51be26abaceb5b6137461cdfe9a57e)
+++ app/src/main/java/com/example/bkkinfoplus/FutarApiClient.java	(revision )
@@ -23,7 +23,7 @@
 /**
  * Created by oli on 2016. 06. 14..
  */
-public class AlertLoader {
+public class FutarApiClient {
 
     private RequestQueue mRequestQueue;
 
@@ -40,7 +40,7 @@
     private HashMap<String, Route> mRoutes;
 
     @Inject
-    public AlertLoader(RequestQueue requestQueue) {
+    public FutarApiClient(RequestQueue requestQueue) {
         // This is for dependency injection
         mRequestQueue = requestQueue;
 
Index: app/src/main/java/com/example/bkkinfoplus/ui/AlertList.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/bkkinfoplus/ui/AlertList.java	(revision )
+++ app/src/main/java/com/example/bkkinfoplus/ui/AlertList.java	(revision )
@@ -0,0 +1,14 @@
+package com.example.bkkinfoplus.ui;
+
+import com.example.bkkinfoplus.Alert;
+
+import java.util.List;
+
+/**
+ * Created by oli on 2016. 06. 20..
+ */
+public interface AlertList {
+    void displayAlerts(List<Alert> alerts);
+
+    void displayNetworkError();
+}
Index: app/src/main/java/com/example/bkkinfoplus/ui/AlertPresenter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/bkkinfoplus/ui/AlertPresenter.java	(revision )
+++ app/src/main/java/com/example/bkkinfoplus/ui/AlertPresenter.java	(revision )
@@ -0,0 +1,12 @@
+package com.example.bkkinfoplus.ui;
+
+/**
+ * Created by oli on 2016. 06. 20..
+ */
+public class AlertPresenter {
+    public void getAlerts() {
+
+    }
+
+
+}
